minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java

```diff
@@ -0,0 +1,165 @@
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.nio.file.StandardOpenOption.WRITE;
+import static org.apache.nifi.minifi.commons.api.MiNiFiConstants.BACKUP_EXTENSION;
+import static org.apache.nifi.minifi.commons.api.MiNiFiConstants.RAW_EXTENSION;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.GZIPOutputStream;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.nifi.c2.client.service.operation.UpdateConfigurationStrategy;
+import org.apache.nifi.controller.FlowController;
+import org.apache.nifi.groups.ProcessGroup;
+import org.apache.nifi.groups.RemoteProcessGroup;
+import org.apache.nifi.minifi.commons.api.FlowEnrichService;
+import org.apache.nifi.services.FlowService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultUpdateConfigurationStrategy implements UpdateConfigurationStrategy {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultUpdateConfigurationStrategy.class);
+
+    private final FlowController flowController;
+    private final FlowService flowService;
+    private final FlowEnrichService flowEnrichService;
+    private final Path flowConfigurationFile;
+    private final Path backupFlowConfigurationFile;
+    private final Path rawFlowConfigurationFile;
+    private final Path backupRawFlowConfigurationFile;
+
+    public DefaultUpdateConfigurationStrategy(FlowController flowController, FlowService flowService, FlowEnrichService flowEnrichService, String flowConfigurationFile) {
+        this.flowController = flowController;
+        this.flowService = flowService;
+        this.flowEnrichService = flowEnrichService;
+        Path flowConfigurationFilePath = Path.of(flowConfigurationFile).toAbsolutePath();
+        this.flowConfigurationFile = flowConfigurationFilePath;
+        this.backupFlowConfigurationFile = Path.of(flowConfigurationFilePath + BACKUP_EXTENSION);
+        String flowConfigurationFileBaseName = FilenameUtils.getBaseName(flowConfigurationFilePath.toString());
+        this.rawFlowConfigurationFile = flowConfigurationFilePath.getParent().resolve(flowConfigurationFileBaseName + RAW_EXTENSION);
+        this.backupRawFlowConfigurationFile = flowConfigurationFilePath.getParent().resolve(flowConfigurationFileBaseName + BACKUP_EXTENSION + RAW_EXTENSION);
+    }
+
+    @Override
+    public boolean update(byte[] rawFlow) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Attempting to update flow with content: \n{}", new String(rawFlow, UTF_8));
+        }
+        try {
+            byte[] enrichedFlowCandidate = flowEnrichService.enrichFlow(rawFlow);
+            stopRootProcessGroup();
```

Comment:
@exceptionfactory, could you help me out here with the review? You are more familiar with the NiFi core functionality, so I would like to double check with you to see if this is the correct way to stop ongoing processing / reload and restart the flow. Thank you in advance